### PR TITLE
Docsite can be built from gradle

### PR DESCRIPTION
**Pull request by Clintonio**
Reason: We're moving away from building docs on the docsite server directly to TC building. Due to a bug in Dokka we can't build on TC, it can be solved multiple ways but I opted to fix it by moving more of the docs building code into gradle so it can be a single gradle step in our TC environment (which would fix it since we can then run the gradle with --no-daemon). In doing so I realised that it could be a gradle module instead of being dumped in the root gradle. From there I reasoned it might be time to remove the docs build directory from the repo since gradle clean was erasing it, and because it causes issues with merges, releasing, etc.

@mikehearn I can always go back a commits and just merge in the original build.gradle only stuff, but I feel this would bring docs more in line with the rest of the project.

### CODE REVIEW

**Comment by mikehearn** [comment]
Now we have docs.corda.net and it shows both master and releases, I agree that it's less critical to have the docs in the repository itself (although, still rather nice for offline working). Perhaps we should offer a zip of the website somewhere.

The robots.txt file can be deleted. It is a holdover from the days before Corda was open source. There's no benefit to blocking search engines from our docsite now, and lots of downside.

**Comment by Clintonio** [comment]
I'll continue this PR today, had a lot of issues to fix in the last couple of days :)

**Comment by Clintonio** [comment]
@mikehearn Probably worth a re-review. I made further changes to related areas (like docs about building docs) and fixing a couple of build order issues.

File: docs/build.gradle [hunk 1]
```diff
@@ -0,0 +1,47 @@
+apply plugin: 'org.jetbrains.dokka'
+apply plugin: 'kotlin'
+
+repositories {
```
Comment (shamsasari) [inline docs/build.gradle]: Haven't we moved this to the root build.gradle?

**Comment by mikehearn** [comment]
Looks OK to me but yes repositories has moved to the root.

**Comment by Clintonio** [comment]
Updated. merge once TC approves.

**Pull request merged**